src/Realm.C

```diff
@@ -4843,6 +4844,27 @@ bool
   return solutionOptions_->activateAdaptivity_;
 }
 
+std::vector<std::string>
+Realm::handle_all_element_part_alias(const std::vector<std::string>& names) const
+{
+  if (names.size() == 1u && names.front() == allElementPartAlias) {
+    std::vector<std::string> new_names;
+    for (const auto* part : meta_data().get_mesh_parts()) {
+      ThrowRequire(part);
+      if (part->topology().rank() == stk::topology::ELEMENT_RANK) {
+        new_names.push_back(part->name());
+      }
+    }
+    return new_names;
+  }
+
+  if (std::find(names.begin(), names.end(), allElementPartAlias) != names.end()) {
+    NaluEnv::self().naluOutputP0() << "Part alias " << allElementPartAlias
+        << " present with other parts; " << allElementPartAlias << " must be a valid mesh part" << std::endl;
+  }
+  return names;
```

Comment:
This brings a question to my mind: Why not let the user leave out `target_name` in the input file and then have `Realm::get_elem_parts_in_mesh()` instead of having this `all_blocks`?

Reply:
Agreed. This could be a nice feature.  It seems logical for IC and materials to apply to the entire domain unless specified.